appservice/src/main/java/nu/yona/server/device/rest/DeviceController.java

```diff
@@ -0,0 +1,161 @@
+public class DeviceController
+{
+	@Autowired
+	private DeviceService deviceService;
+
+	@Autowired
+	private UserService userService;
+
+	@Autowired
+	private CurieProvider curieProvider;
+
+	@RequestMapping(value = "/", method = RequestMethod.GET)
+	@ResponseBody
+	public HttpEntity<Resources<DeviceResource>> getAllDevices(@RequestHeader(value = PASSWORD_HEADER) Optional<String> password,
+			@PathVariable UUID userId)
+	{
+		try (CryptoSession cryptoSession = CryptoSession.start(password, () -> userService.canAccessPrivateData(userId)))
+		{
+			return new ResponseEntity<>(
+					createAllDevicesCollectionResource(curieProvider, userId, deviceService.getDevicesOfUser(userId)),
+					HttpStatus.OK);
+		}
+	}
+
+	@RequestMapping(value = "/{buddyId}", method = RequestMethod.GET)
+	@ResponseBody
+	public HttpEntity<DeviceResource> getDevice(@RequestHeader(value = PASSWORD_HEADER) Optional<String> password,
+			@PathVariable UUID userId, @PathVariable UUID deviceId)
+	{
+		try (CryptoSession cryptoSession = CryptoSession.start(password, () -> userService.canAccessPrivateData(userId)))
+		{
+
+			return createOkResponse(userId, deviceService.getDevice(deviceId));
+		}
+	}
+
+	private HttpEntity<DeviceResource> createOkResponse(UUID userId, DeviceBaseDto device)
+	{
+		return createResponse(userId, device, HttpStatus.OK);
+	}
+
+	private HttpEntity<DeviceResource> createResponse(UUID userId, DeviceBaseDto device, HttpStatus status)
+	{
+		return new ResponseEntity<>(new DeviceResourceAssembler(curieProvider, userId).toResource(device), status);
+	}
+
+	public static Resources<DeviceResource> createAllDevicesCollectionResource(CurieProvider curieProvider, UUID userId,
+			Set<DeviceBaseDto> allBuddiesOfUser)
+	{
+		return new Resources<>(new DeviceResourceAssembler(curieProvider, userId).toResources(allBuddiesOfUser),
+				getAllDevicesLinkBuilder(userId).withSelfRel());
+	}
+
+	static ControllerLinkBuilder getAllDevicesLinkBuilder(UUID userId)
+	{
+		DeviceController methodOn = methodOn(DeviceController.class);
+		return linkTo(methodOn.getAllDevices(null, userId));
+	}
+
+	public static ControllerLinkBuilder getDeviceLinkBuilder(UUID userId, UUID deviceId)
+	{
+		DeviceController methodOn = methodOn(DeviceController.class);
+		return linkTo(methodOn.getDevice(Optional.empty(), userId, deviceId));
+	}
+
+	static class DeviceResource extends Resource<DeviceBaseDto>
+	{
+		private final CurieProvider curieProvider;
+		private final UUID userId;
+
+		public DeviceResource(CurieProvider curieProvider, UUID userId, DeviceBaseDto buddy)
+		{
+			super(buddy);
+			this.curieProvider = curieProvider;
+			this.userId = userId;
+		}
+	}
+
+	static class DeviceResourceAssembler extends ResourceAssemblerSupport<DeviceBaseDto, DeviceResource>
+	{
+		private final UUID userId;
+		private final CurieProvider curieProvider;
+
+		public DeviceResourceAssembler(CurieProvider curieProvider, UUID userId)
+		{
+			super(BuddyController.class, DeviceResource.class);
```

Comment:
`DeviceController` instead of `BuddyController`

Reply:
Done